### PR TITLE
Feat/24/chatgpt interact

### PR DESCRIPTION
**Pull request by matt-newhall**

### CODE REVIEW

File: src/app/pages/api/openai.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  const response = await fetch('https://api.openai.com/v1/chat/completions', {
+    method: 'POST',
+    headers: {
+      'content-type': 'application/json',
+      Authorization: 'Bearer ' + process.env.API_KEY,
+    },
+    body: JSON.stringify({
+      model: 'gpt-3.5-turbo',
+      messages: [{ role: 'user', content: 'Say this is a test!' }],
+      temperature: 0.7,
```
Comment (will249) [inline src/app/pages/api/openai.ts:13]: Is this is a default temperature value?

Comment (matt-newhall) [reply src/app/pages/api/openai.ts:13]: no specific reason for 0.7, I think this was an example from the docs, moreso to show as an example of where to pass it in

File: src/app/pages/api/openai.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import { OpenaiApiType } from '@/app/types/OpenaiApiType';
+
+const fetchData = async (): Promise<OpenaiApiType> => {
+  const response = await fetch('https://api.openai.com/v1/chat/completions', {
```
Comment (will249) [inline src/app/pages/api/openai.ts:4]: Do you think we should decide on a standard re: fetching from api vs using openai sdk?

File: src/app/types/OpenaiApiType.ts
Comment (will249) [inline src/app/types/OpenaiApiType.ts:1]: Nice!

File: src/app/pages/api/openai.ts (hunk 1 above)
Comment (will249) [reply src/app/pages/api/openai.ts:13]: Ok cool, we can keep a note of it for experimentation phase

File: src/app/pages/api/openai.ts (hunk 2 above)
Comment (matt-newhall) [reply src/app/pages/api/openai.ts:4]: opted for fetching from API for this ticket to not include the extra dependency but open to challenges on this !

Comment (will249) [reply src/app/pages/api/openai.ts:4]: Cool, makes sense within next app, can still discuss for our one-shot code

**Review by will249 — APPROVED** [review]

**Pull request merged**